xgcm/grid.py

```diff
@@ -645,7 +662,7 @@ def diff(
             boundary_discontinuity,
             vector_partner,
             keep_coords=keep_coords,
-        )
+        )*self.direction_sign
```

Comment:
IMO, I think we do want it in `diff`. Otherwise, what's the point of having the parameter at all? If you don't want to use it, just set `positive_upward=True` (the default).

I would also prefer to implement this in the following way.
```python
result = self._neighbor_binary_func(...)
if not self.positive_upward:
    result *= -1
return result
```

Although it doesn nothing, needlessly multiplying by 1 increases the computational cost of the dask graph.

Reply:
>  If you don't want to use it, just set positive_upward=True (the default).

I favor opt-in. But that's the opposite interpretation of `positive_upward`, that means positive values in the result indicate field values increasing upward in physical space. Maybe we need something else then.

Reply:
Agree `positive_upward` may not be the right term. For the sake of discussion, let's just call it, super verbosely,`logical_ordering_is_the_same_as_physical_ordering=True`. The value of `True` represents the current default, where we make no distinction between the logical and physical ordering.

I'm unclear on what you mean by opt-out.  With `logical_ordering_is_the_same_as_physical_ordering=True`, the default, nothing would change from the current behavior. With `numerical_ordering_is_the_same_as_physical_ordering=False`, the `diff` and `derivative` calls would get an extra factor of -1. By setting this parameter to a non-default value, isn't the user already "opting in"? Or do you imagine they would specify numerical_ordering_is_the_same_as_physical_ordering=False` when creating a `Grid` and then again when calling `diff`? That feels a bit redundant.

Reply:
>  Or do you imagine they would specify numerical_ordering_is_the_same_as_physical_ordering=Falsewhen creating aGridand then again when callingdiff`? That feels a bit redundant.

No. I am saying the same thing as you but with inverted perspective =) . `use_physical_ordering` seems like a good name. I'll switch the cf_xarray flag to match whatever we decide here.

cc @malmans2 who may have ideas.

Reply:
I think @rabernat s implementation is good. We don't want to grow any dask graphs hahaha.

I am not super convinced of using `physical_ordering` as a kwarg. I think it very nicely describes what is happening to us devs, but might be very confusing to potential users. I did not dislike `direction` to much TBH, but maybe the arguments could be `forward`/`backward` (`forward = logical ordering represents increasing coordinates`). Meh, not in love with that. Lets discuss this further...

Reply:
Thank you for all of these helpful comments. I likewise agree with @rabernat 's implementation, and will make those changes.

In terms of naming, it seems tricky to find something that is both logically correct and intuitive to the user. I like the use of "ordering". I was thinking about `ordered=True` or `consistent_ordering=True`? Not entirely logically correct perhaps, but close enough. Together with a careful description in the docstring this might be enough?

Reply:
Regarding naming, MITgcm uses [rkSign](https://github.com/MITgcm/MITgcm/blob/b7589390f219db12d2062055ad7f822bd7b80a0e/model/inc/GRID.h#L308-L309). Maybe we could use `axis_sign`?

Reply:
I like that!

Reply:
With the options then being `positive` for the case that the logical ordering matches the physical ordering, and `negative` for the opposite? So in the common case for the vertical axis the sign would be negative? I can see a user getting confused by that implementation, just in the same sense that `positive_upward` is not completely foolproof.  That also gets away from the boolean implementation that @dcherian favored.

Reply:
Isn't `axis_sign` (or any other kwarg) useless if it isn't boolean? If you're going to specify `axis_sign=-1` you could just do `-1 * grid.diff(...)`.

Reply:
I wanted to revive this thread here, because I think this is some nice functionality that can be implemented relatively easily. 

After rereading this I think I am in favor of 'axis_sign` after all, mostly because it is succinct. I think the most important thing to do is adding good docstrings and examples to make it really clear what is happening.